quic: flaky TestConnectionGating/accepted_connections

```
  === RUN   TestConnectionGating/accepted_connections
      conn_test.go:306: 
          	Error Trace:	conn_test.go:306
          	Error:      	Received unexpected error:
          	            	Application error 0x47415445: connection gated
          	Test:       	TestConnectionGating/accepted_connections
      conn_test.go:298: 
          	Error Trace:	conn_test.go:298
          	            				asm_amd64.s:1581
          	Error:      	Received unexpected error:
          	            	server closed
          	Test:       	TestConnectionGating/accepted_connections
  === RUN   TestConnectionGating/secured_connections
  --- FAIL: TestConnectionGating (0.01s)
      --- FAIL: TestConnectionGating/accepted_connections (0.00s)
      --- PASS: TestConnectionGating/secured_connections (0.01s)

--- a/p2p/transport/quic/conn_test.go
+++ b/p2p/transport/quic/conn_test.go
@@ -303,9 +303,11 @@ func TestConnectionGating(t *testing.T) {
 		defer clientTransport.(io.Closer).Close()
 		// make sure that connection attempts fails
 		conn, err := clientTransport.Dial(context.Background(), ln.Multiaddr(), serverID)
-		require.NoError(t, err)
-		_, err = conn.AcceptStream()
-		require.Error(t, err)
+		// In rare instances, the connection gating error will already occur on Dial.
+		// In most cases, it will be returned by AcceptStream.
+		if err == nil {
+			_, err = conn.AcceptStream()
+		}
 		require.Contains(t, err.Error(), "connection gated")
 
 		// now allow the address and make sure the connection goes through

